### PR TITLE
TCP Requests

### PR DESCRIPTION
**Pull request by k3z0**
Max Requests-per-Second

### CODE REVIEW

File: src/EmbedIO/Security/IIPBanningCriterion.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Net;
+using System.Threading.Tasks;
+
+namespace EmbedIO.Security
+{
+    public interface IIPBanningCriterion
```
Comment (geoperez) [inline src/EmbedIO/Security/IIPBanningCriterion.cs]: Xmldoc!

File: src/EmbedIO/Security/IPBanningModule.cs [hunk 2]
```diff
@@ -8,152 +8,160 @@
...
+        /// <param name="maxRequestPerSecond">The maximum requests per second.</param>
+        /// <param name="maxMatchCount">The maximum number of regex matches before banning an IP.</param>
+        /// <param name="secondsMatchingPeriod">The matching period.</param>
         public IPBanningModule(string baseRoute,
             IEnumerable<string> failRegex,
-            int banTime = DefaultBanTime,
-            int maxRetry = DefaultMaxRetry)
-            : this(baseRoute, failRegex, null, banTime, maxRetry)
+            int banMinutes = DefaultBanMinutes,
+            int maxRequestPerSecond = DefaultMaxRequestsPerSecond,
```
Comment (geoperez) [inline src/EmbedIO/Security/IPBanningModule.cs]: ?? Do we need this here?

File: src/EmbedIO/Security/IPBanningRequestsCriterion.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EmbedIO.Security
+{
+    public class IPBanningRequestsCriterion : IIPBanningCriterion
```
Comment (geoperez) [inline src/EmbedIO/Security/IPBanningRequestsCriterion.cs]: Xmldoc

**Comment by geoperez** [comment]
@k3z0 please complete the documentation and unit testing.

File: src/EmbedIO/Security/IPBanningRequestsCriterion.cs (hunk 3 above)
Comment (k3z0) [reply src/EmbedIO/Security/IPBanningRequestsCriterion.cs]: It is done

**Pull request merged**